Bottom: 2cd2531779446444de24991b34f35e1b50b0972b
Top:    649a3279adb311dec84a64134482a70b91571d4f
Author: Robert Collins <robertc@robertcollins.net>
Date:   2013-02-24 00:06:01 +1300

Refresh of extended-to-stream

---

diff --git a/testtools/testresult/real.py b/testtools/testresult/real.py
index 0357e18..c46150e 100644
--- a/testtools/testresult/real.py
+++ b/testtools/testresult/real.py
@@ -1159,14 +1159,14 @@ class ExtendedToStreamDecorator(CopyStreamResult, StreamSummary, TestControl):
             for name, content in details.items():
                 mime_type = repr(content.content_type)
                 for file_bytes in content.iter_bytes():
-                    self.file(name, file_bytes,
+                    self.status(file_name=name, file_bytes=file_bytes,
                         mime_type=mime_type, test_id=test_id, timestamp=now)
-                self.file(name, _b(""), eof=True,
+                self.status(file_name=name, file_bytes=_b(""), eof=True,
                     mime_type=mime_type, test_id=test_id, timestamp=now)
         if reason is not None:
-            self.file('reason', reason.encode('utf8'), eof=True,
-                mime_type="text/plain; charset=utf8", test_id=test_id,
-                timestamp=now)
+            self.status(file_name='reason', file_bytes=reason.encode('utf8'),
+                eof=True, mime_type="text/plain; charset=utf8",
+                test_id=test_id, timestamp=now)
         self.status(test_id, status, test_tags=self.current_tags,
             timestamp=now)
 
diff --git a/testtools/tests/test_testresult.py b/testtools/tests/test_testresult.py
index 1b69a86..56a84a6 100644
--- a/testtools/tests/test_testresult.py
+++ b/testtools/tests/test_testresult.py
@@ -713,6 +713,10 @@ class TestExtendedToStreamDecorator(TestCase):
              None,
              True,
              None,
+             None,
+             False,
+             None,
+             None,
              now),
             ('status',
              'testtools.tests.test_testresult.TestExtendedToStreamDecorator.test_explicit_time',
@@ -720,6 +724,10 @@ class TestExtendedToStreamDecorator(TestCase):
               set(),
               True,
               None,
+              None,
+              False,
+              None,
+              None,
               now),
              ('stopTestRun',)], log._events)
